### ISSUE TITLE
Poor performance in SQL notebooks with 30 resultsets

### ISSUE THREAD

**Issue by vickyharp** [microsoft/azuredatastudio#7716]
Issue Type: <b>Bug</b>

Render performance and particularly scroll performance is terrible with a SQL notebook running this cell:

``` 
select top 10 * from sys.databases  
go 30
```

Azure Data Studio version: azuredatastudio-insiders 1.12.0-insider (93c9426f25e632e8e3dd6e572d48d559ae07a7bf, 2019-10-10T18:18:16.586Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7600U CPU @ 2.80GHz (4 x 2904)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.84GB (2.23GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
admin-tool-ext-win|Mic|0.0.2
big-data-cluster|Mic|0.0.1
cms|Mic|0.5.0
dacpac|Mic|0.7.0
import|Mic|0.11.0
powershell|ms-|2019.9.0
profiler|Mic|0.10.0
query-history|Mic|0.1.0
schema-compare|Mic|0.7.0
sql-vnext|Mic|0.14.1
azdata-sanddance|msr|1.3.7

(1 theme extensions excluded)

</details>

**Comment by SQLDBAWithABeard** [comment]
Also the Glenn Berry Diagnostic Query Notebook has significant poor performance when opening a Notebook that has been scheduled (ie already has the results) and if you choose to Run All Cells. When we first created these I didnt notice this.

**Comment by MsSQLGirl** [comment]
@chlafreniere - bubbling this up. 







- Azure Data Studio Version:
Version: 1.20.0-rc1 (user setup)
Commit: c36961378259f18fb3af0257d89c482fc3afe1a0
Date: 2020-07-11T13:14:59.340Z
VS Code: 1.46.0
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041

- Azure Data Studio Version:
Version: 1.20.0-insider (user setup)
Commit: 7d5ecfe905e2e521e9c10c8ce8ffca881bc8c611
Date: 2020-07-10T05:33:19.580Z
VS Code: 1.46.0
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041


Steps to Reproduce:

1. Launch Azure Data Studio and load Glenn Berry's SQL 2019 Diagnostic Information Query from [here](https://www.dropbox.com/s/h2kbdxdu7hcnsam/SQL%20Server%202019%20Diagnostic%20Information%20Queries.ipynb?dl=0).
2. Execute the notebook against your SQL 2019 environment. This will take a few minutes.
3. Save the notebook and close it.
4. Reopen the notebook (which has pre-executed info). After 30 seconds, it will show this error:
![image](https://user-images.githubusercontent.com/6405919/87253170-cec7e280-c42d-11ea-9b9a-3df4cf7bd4ad.png)
5. After waiting for a few minutes, it would eventually load (ignoring the dialog box). When it loads, please click on "Keep Waiting" (don't choose Close or Reopen otherwise it will try to reload ADS and would have to start back from Step 4).
6. When it does load, it is generally unresponsive.
* Try Find "Query 47" on the notebook, it will be fairly slow (a few seconds to find it). 
* Scrolling up and down the notebook is also very slow. 
* Trying to click on a text cell to do editing is also unresponsive.
* Trying to add a new cell to execute a new code is also very slow.

Note: Query 47 has fairly lengthy result set. I was able to load / render the notebook in Azure Dev Ops :) 
![image](https://user-images.githubusercontent.com/6405919/87253142-8f00fb00-c42d-11ea-9922-99020edc70cf.png)

Other queries may also return lengthy result sets; which vary from one SQL Server environment to another.

**Comment by MsSQLGirl** [comment]
@lucyzhang929 - can you please confirm that the latest fix in Insiders v1.32.0 would improve perf for the issue that @vickyharp reported above? Thank you.

**Comment by MsSQLGirl** [comment]
@SQLDBAWithABeard - w.r.t Glenn Berry Diagnostic Query Notebook... 
We have an update on loading time improvement.

Follow the steps below to try out the improved loading time:

1. Download Azure Data Studio Insider build: v1.32.0-insider.
Version: 1.32.0-insider (user setup)
Commit: 991b9af19880b20f5382daf5249aa31cc919560b
Date: 2021-08-06T05:23:36.445Z

2. Launch Azure Data Studio and go to Settings ⚙.

3. Disable Results Grid: Auto Size Columns.
![Auto Size Columns](https://www.mssqlgirl.com/wp-content/uploads/2021/08/AutoSizeColumns.png)

4. Enable Notebook: Enable Incremental Grid Rendering.
![Enable Incremental Grid Rendering](https://www.mssqlgirl.com/wp-content/uploads/2021/08/EnableIncrementalGridRendering.png)

Can you please let us know what you think?

Thank you!

**Comment by SQLDBAWithABeard** [comment]
Awesome - That works for me :-)

Normal ADS
![image](https://user-images.githubusercontent.com/6729780/128860917-29dd76c5-c190-4ba7-bc61-ca6d2436127c.png)

Insiders

![image](https://user-images.githubusercontent.com/6729780/128861066-7e64432e-45c3-4a2d-85a4-c446879e4e0f.png)

**Comment by lucyzhang929** [comment]
@MsSQLGirl The 'Enable Notebook: Enable Incremental Grid Rendering' setting in Insiders is targeted for initial loading time when opening large notebooks and does not affect when a cell is run. However, turning off the 'Auto Size Columns' setting will improve the perf issue @vickyharp mentioned. I'm currently working on perf improvements for the Auto Size Columns feature.

**Comment by SQLDBAWithABeard** [comment]
Yup I was able to open in Insiders and do a run all on the Glen Berry Notebook which would have crashed it before
